### PR TITLE
Unify PETSc and Firedrake docker images

### LINKED ISSUE #86: Unify Docker images?
While working on #83 and #85 I realised we're quite inefficient with our Docker management.

May I ask what was the idea behind separating PETSc and Firedrake images? Since we only use the PETSc image as a base for the Firedrake one, and we always deploy them at the same time, that is quite inefficient. I think it would be much better to unify them into one Dockerfile with a multi-stage build. Keeping them separated would make sense to me if, for example, we'd update the PETSc image less frequently than the Firedrake one (which might be sensible too?).

Also, installing UM2N and its dependencies now takes only about 20 seconds and ~1.5GB (which could probably be cut down too by cleaning up its dependencies, such as jupyter). But here it's more clear why it's separated

### PR DESCRIPTION
**Pull request by ddundo**
Closes #86.

Merging into #85 so to not trigger the build multiple times on the main branch.

@jwallwork23 after this new firedrake-parmmg image is pushed, in a new PR I'd make the UM2N container use it as a base image, since it's smaller than firedrake-vanilla. Then its size also drops from 10.23GB to 6.68GB. Does that sound good?

### CODE REVIEW

**Review by jwallwork23 — APPROVED** [review]
This all looks good, thanks. I've successfully built the Docker image locally and can confirm it works. Just one minor comment.

I can confirm that the image is still 5.1GB when implemented this way.

File: .github/workflows/docker_firedrake-parmmg.yml [hunk 1]
```diff
@@ -15,13 +15,9 @@ on:
...
-  # Trigger this workflow when docker_petsc-parmmg.yml completes on the main branch
-  workflow_run:
-    workflows: ['Build bespoke PETSc Docker container']
-    types:
-      - completed
-    branches:
-      - main
+   # Build and push the Docker container every Saturday at 12AM
+  schedule:
+    - cron: '0 0 * * 6'
```
[the remark this answers is not in the document]
Comment (ddundo) [reply .github/workflows/docker_firedrake-parmmg.yml:20]: The test suites are run on Sunday :) I always mix this up too

**Comment by ddundo** [comment]
Since review, minor change in d688d35b4c0c783c6712ab5c1b2149890c78e199 that shaves off another 100MB. It's minor and I checked locally so I cancelled the CI here and will go ahead and merge

**Comment by ddundo** [comment]
@jwallwork23 are you happy for me to delete [petsc-parmmg](https://github.com/orgs/mesh-adaptation/packages/container/package/petsc-parmmg)?

**Comment by jwallwork23** [comment]
> @jwallwork23 are you happy for me to delete [petsc-parmmg](https://github.com/orgs/mesh-adaptation/packages/container/package/petsc-parmmg)?

Sure, I guess we don't need it anymore.

**Pull request merged**